core/src/main/scala/kafka/coordinator/group/GroupMetadata.scala

```diff
@@ -127,6 +127,17 @@ case class GroupSummary(state: String,
                         members: List[MemberSummary])
 
 /**
+  * We cache offset commits along with their commit record offset. This enables us to ensure that the latest offset
+  * commit is always materialized when we have a mix of transactional and regular offset commits. Without preserving
+  * information of the commit record offset, compaction of the offsets topic it self may result in the wrong offset commit
+  * being materialized.
+  */
+case class CommitRecordMetadataAndOffset(commitRecordOffset: Long,
```

Comment:
By the way, it seems our assumption is that we would never have two offset commits for the same partition as part of the same batch. This seems to be guaranteed by the schema of the OffsetCommitRequest, but maybe we should mention the assumption in a comment?